src/lsm/manifest_level_fuzz.zig

```diff
@@ -0,0 +1,426 @@
+            var env: Environment = undefined;
+
+            const node_pool_size = ManifestLevel.Keys.node_count_max +
+                ManifestLevel.Tables.node_count_max;
+            env.pool = try NodePool.init(allocator, node_pool_size);
+            defer env.pool.deinit(allocator);
+
+            env.level = try ManifestLevel.init(allocator);
+            defer env.level.deinit(allocator, &env.pool);
+
+            env.buffer = TableBuffer.init(allocator);
+            defer env.buffer.deinit();
+
+            env.tables = TableBuffer.init(allocator);
+            defer env.tables.deinit();
+
+            env.random = random;
+            env.snapshot = 1; // the first snapshot is reserved.
+
+            for (fuzz_ops) |fuzz_op, op_index| {
+                log.debug("Running fuzz_ops[{}/{}] == {}", .{ op_index, fuzz_ops.len, fuzz_op });
+                switch (fuzz_op) {
+                    .insert_tables => |amount| try env.insert_tables(amount),
+                    .update_tables => |amount| try env.update_tables(amount),
+                    .take_snapshot => try env.take_snapshot(),
+                    .remove_invisible => |amount| try env.remove_invisible(amount),
+                    .remove_visible => |amount| try env.remove_visible(amount),
+                }
+            }
+        }
+
+        fn insert_tables(env: *Environment, amount: usize) !void {
+            assert(env.buffer.items.len == 0);
+
+            // Generate random, non-overlapping TableInfo's into env.buffer:
+            {
+                var insert_count: usize = 0;
+                const insert_max = @minimum(amount, table_count_max - env.level.keys.len());
+                var key = env.random.uintAtMostBiased(Key, table_count_max * 64);
+
+                while (insert_count < insert_max) : (insert_count += 1) {
+                    const table = env.generate_non_overlapping_table(key);
+                    try env.buffer.append(table);
+                    key = table.key_max;
+                }
+            }
+
+            const tables = env.buffer.items;
+            defer env.buffer.clearRetainingCapacity();
+
+            // Insert the generated tables into the ManifestLevel:
+            for (tables) |*table| {
+                assert(table.visible(env.snapshot));
+                assert(table.visible(lsm.snapshot_latest));
+                env.level.insert_table(&env.pool, table);
+            }
+
+            // Insert the generated tables into the Environment for reference:
+            for (tables) |*table| {
+                const index = binary_search.binary_search_values_raw(
+                    Key,
+                    TableInfo,
+                    key_min_from_table,
+                    compare_keys,
+                    env.tables.items,
+                    table.key_max,
+                    .{},
+                );
+
+                // Can't be equal as the tables may not overlap
```

Comment:
Comments are sentences and should always end with punctuation.

For example, either `.` if they refer clearly to the next line in isolation, or `:` if they refer to multiple lines beneath.